File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if (list1 == null)
        {
            return list2;
        }
        if (list2 == null)
        {
            return list1;
        }
        ListNode ls1 = list1;
        ListNode ls2 = list2;
        ListNode head = null;
        ListNode tail = null;
        if (ls1.val <= ls2.val)
        {
            head = ls1;
            tail = ls1;
            ls1 = ls1.next;
        }
        else
        {
            head = ls2;
            tail = ls2;
            ls2 = ls2.next;
        }
        while (ls1 != null && ls2 != null)
        {
            if (ls1.val <= ls2.val)
            {
                tail.next = ls1;
                tail = ls1;
                ls1 = ls1.next;
            }
            else
            {
                tail.next = ls2;
                tail = ls2;
                ls2 = ls2.next;
            }
        }
        if (ls1 == null)
        {
            tail.next = ls2;
        }
        else
        {
            tail.next = ls1;
        }
        return head;
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if list1 == None:
            return list2
        if list2 == None:
            return list1
        curr = list1
        curr2 = list2
        if curr.val <= curr2.val:
            temp = curr
            head = temp
            curr = curr.next
        else:
            temp = curr2
            head = temp
            curr2 = curr2.next
        while curr != None and curr2 != None:
            if curr.val <= curr2.val:
                temp.next = curr
                temp = curr
                curr = curr.next
            else:
                temp.next = curr2
                temp = curr2
                curr2 = curr2.next
        if curr != None:
            temp.next = curr
        if curr2 != None:
            temp.next = curr2
        return head